oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -340,6 +365,71 @@ public String getQuotaProjectId() {
     return quotaProjectId;
   }
 
+  /**
+   * Clone the UserCredential with the specified scopes.
+   *
+   * <p>Should be called before use for instances with empty scopes.
+   */
+  @Override
+  /*  public GoogleCredentials createScoped(Collection<String> newScopes) {
+
+    this.scopes = scopes;
+    return this;
+  }*/
+
+  public GoogleCredentials createScoped(Collection<String> newScopes) {
+    return new UserCredentials(
+        clientId,
+        clientSecret,
+        refreshToken,
+        getAccessToken(),
+        newScopes,
+        transportFactory,
+        tokenServerUri,
+        quotaProjectId);
+  }
+
+  /**
+   * Returns a Google ID Token from the user credential
+   *
+   * @param targetAudience currently unused for UserCredential.
+   * @param options list of Credential specific options for for the token. Currently unused for
+   *     UserCredentials.
+   * @throws IOException if the attempt to get an IdToken failed
+   * @return IdToken object which includes the raw id_token and expiration
+   */
+  @Beta
+  @Override
+  public IdToken idTokenWithAudience(String targetAudience, List<Option> options)
+      throws IOException {
+    JsonFactory jsonFactory = OAuth2Utils.JSON_FACTORY;
+
+    GenericData tokenRequest = new GenericData();
+    tokenRequest.set("grant_type", GRANT_TYPE);
+    tokenRequest.set("client_id", GOOGLE_CLIENT_ID);
+    tokenRequest.set("client_secret", GOOGLE_CLIENT_SECRET);
+    tokenRequest.set("refresh_token", this.refreshToken);
+    // build scope value
+    Iterator<String> it = this.scopes.iterator();
+    String customScopes = it.next();
+    while (it.hasNext()) {
+      customScopes += "+" + it.next();
+    }
+    tokenRequest.set("scope", customScopes);
```

Comment:
I don't see any documentation for this parameter against `https://oauth2.googleapis.com/token`

Reply:
Agree, the scope parameters is not documented -> https://developers.google.com/identity/protocols/oauth2/web-server#offline

But required to scope correctly the generated token.

Reply:
@chingor13 I don't know what to say! The request need it to work. 

Of course, it's not documented, and I don't know why, maybe a mistake in the current documentation?

Reply:
Possible we need to update the docs.

Reply:
I found the scope parameter on https://developers.google.com/identity/protocols/oauth2/web-server#offline

 Make sure you're looking at the HTTP/RES documents and not at one of the language specific tabs.